### PR TITLE
fix(server) fix abrupt stop

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?
Fix: https://github.com/oven-sh/bun/issues/12489

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

Added a test

### CODE REVIEW

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -2632,6 +2632,11 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
             ctx.setAbortHandler();
         }
 
+        fn isAborted(this: *RequestContext) bool {
```
Comment (Jarred-Sumner) [inline src/bun.js/api/server.zig]:
```suggestion
        fn isAborted(this: *const RequestContext) bool {
```

File: src/bun.js/api/server.zig [hunk 2]
```diff
@@ -1561,22 +1561,19 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
...
+
             const resp = ctx.resp.?;
+            resp.clearAborted();
             const has_responded = resp.hasResponded();
             if (!has_responded)
                 ctx.runErrorHandler(
                     value,
                 );
 
-            if (ctx.isAborted()) {
```
Comment (Jarred-Sumner) [inline src/bun.js/api/server.zig]: What if the request aborts after the error handler runs?

Comment (cirospaciari) [reply src/bun.js/api/server.zig]: fixed https://github.com/oven-sh/bun/pull/12472/commits/c7bcbe8d6835f06b122ae22f7081098d84064a6d

File: packages/bun-uws/src/HttpResponse.h [hunk 3]
```diff
@@ -581,6 +581,12 @@ struct HttpResponse : public AsyncSocket<SSL> {
         return this;
     }
 
+    HttpResponse* clearOnAborted() {
+        HttpResponseData<SSL> *httpResponseData = getHttpResponseData();
+
+        httpResponseData->onAborted = nullptr;
```
Comment (Jarred-Sumner) [inline packages/bun-uws/src/HttpResponse.h:593]:
```suggestion
        httpResponseData->onAborted = nullptr;
        httpResponseData->onWritable = nullptr;        
```

Comment (cirospaciari) [reply packages/bun-uws/src/HttpResponse.h:593]: fixed in https://github.com/oven-sh/bun/pull/12472/commits/6094340309d0fddecd377aeabb706d04d516afed we clean both only in some cases (onWritable being set or not changes some internal behaviors)

**Pull request merged**